### PR TITLE
NMS-14134: backport spring introspection cache fixes from Spring 5.3.18

### PR DESCRIPTION
**Pull request by RangerRick**
This PR is a light/smaller version of [my unfinished work blowing up Spring](https://github.com/OpenNMS/opennms/pull/4480).  It does not attempt to rip out versions of Spring other than 4.2, or attempt to do elaborate work to clean up the features list, and instead creates patches for each versions of Spring provided in the Karaf `spring-legacy` features XML.

Like the other PR, I tried to break the commits into logical/easy-to-digest bits.
The ones labeled `NMS-14134` are directly related to manipulating Spring and Karaf.
The ones that _aren't_ prefixed with `NMS-14134` are smaller general-use support commits that made it easier to do the former.

* backport the enabling of the `maven-enforcer-plugin` to foundation-2019 to make it easy to block the bad `spring-beans` dependencies
* fix a few small dependency bugs that caused build failures or simplified the dependency tree
* fix the `runInPlace.sh` scripts to make it easier to test Minion and Sentinel Karaf runs
* fix the `paxWebVersion` property to match the version of pax-web supplied by Karaf 4.2.6 (this must have been an oversight in previous updates)
* clean up the `custom.properties` distributed with the OpenNMS core to actually match the Spring jars in the classpath (from `$OPENNMS_HOME/lib`)
* bump the Spring 4.3 version to the latest (last) version: `4.3.30.RELEASE`
* completely remove Spring 5.x from the runtime, we don't use it for anything and our own code and embedded dependencies are happy to run in Spring 4.x
* create shaded versions of `spring-beans` 3.1 through 4.3
* change our `spring-legacy` to use the shaded versions of `spring-beans`, and fix any build stuff necessary to make that run smoothly

### External References

* JIRA (Issue Tracker): http://issues.opennms.org/browse/NMS-14134

### CODE REVIEW

File: container/shared/pom.xml [hunk 1]
```diff
@@ -68,25 +72,11 @@
                         <feature>eventadmin</feature>
                     </startupFeatures>
                     <bootFeatures>
-                        <feature>wrap</feature>
```
Comment (cgorantla) [inline container/shared/pom.xml:71]: Are all these features part of karaf standard features ?

**Review by cgorantla — COMMENTED** [review]
I went through each commit.
Risk is probably breaking something in karaf that is not covered by tests. Hopefully nothing  unknown.

Looks good to me!

File: container/shared/pom.xml (hunk 1 above)
Comment (RangerRick) [reply container/shared/pom.xml:71]: Yup! It was silly to list them manually, it was literally the exact same list in the exact same order.

**Comment by RangerRick** [comment]
> I went through each commit. Risk is probably breaking something in karaf that is not covered by tests. Hopefully nothing unknown.
> 
> Looks good to me!

Yeah, next step after merging is getting Cloud team and support to help me with some heavier testing. That said, I did a bunch of manual testing of minion and sentinel (where the risk was greater) and things seem to all be loading just fine.

**Pull request merged**